### DISCUSSION TITLE
Keenetic OS

### DISCUSSION THREAD

**Discussion by snagovskiy** [Ground-Zerro/DomainMapper#64] [Q&A] [answered]
Что то не пойму как добавить маршруты в ротуер если выбрать сохранения файла для кинетика. Для win добавляется так как bat. Но мне больше подходит для кинетика т.к. можно добавить описание маршутов.

**Comment by Ground-Zerro** [comment, marked as answer]
@snagovskiy 
Для добавления маршрутов через **WEB** интерфейс используется **WIN** формат.

https://github.com/user-attachments/assets/dc927194-e71e-4b74-ba7f-b6ce6a868fd1

Формат **keenetic** предназначен для добавления через **CLI** интерфейс. Для этого нужно подключиться к роутеру терминалом по SSH на 22 порт, предварительно установив эту службу в компонентах и настроив доступ пользователю.

**Reply by snagovskiy** [reply]
Спасибо! Через web и добавлял по win только расширение .bat менял файла, скачал последнюю версию 0.1.3 увидел отдельно для кинетика и стало интересно как добавить его, спасибо за подсказку SSH, буду разбираться как.

**Comment by Turbo198202** [comment]
Здравствуйте! Извините если я не туда пишу. Не могли бы вы добавить в батник Domain Mapper
вот эти сайты https://kinoplay.app https://kino.pub https://filmix.me https://cub.red https://hdrezka.me https://rufilmtv.tv
И если можно дайте мне пожалуйста подробную инструкцию, что нужно установить, какие команды и куда нужно писать, чтоб я сам для себя мог добавлять какие либо сайты.. Спасибо большое!!!

**Reply by snagovskiy** [reply]
Автор же сделал возможным добавлять кастомные домены. Почитайте инструкцию

**Reply by Turbo198202** [reply]
Извините не могу найти. Дайте пожалуйста ссылку.

**Reply by snagovskiy** [reply]
> Извините не могу найти. Дайте пожалуйста ссылку.

https://github.com/Ground-Zerro/DomainMapper/blob/main/README.md

**Reply by Ground-Zerro** [reply]
@Turbo198202 
> куда нужно писать, чтоб я сам для себя мог добавлять какие либо сайты.

- Создайте файл `custom-dns-list.txt`, запишите в него доменные имена и разместите его в одной папке  со скриптом (или батником, если используется он). Скрипт или батник нужно скачать к себе на ПК, а не запускать с GitHub.
Список будет подхвачен при запуске и появится в меню как **"Custom DNS list"**.

- Пример файла `custom-dns-list.txt`:
```
ab.chatgpt.com
api.openai.com
arena.openai.com
```
Указание URL вместо доменного имени **НЕ ДОПУСКАЕТСЯ** (например, `ab.chatgpt.com/login` или `https://ab.chatgpt.com/`).

**Comment by Sam4015** [comment]
А будет ли возможность сохранять сконфигурированный в   Domainmapper файл в формате для hydraroute, было бы очень удобно заменить файл сконфигурированный и все

**Reply by Ground-Zerro** [reply]
> сконфигурированный в Domainmapper файл в формате для hydraroute

Используется разный синтаксис.
в DM одна строка - один домен
в HR одна строка - домены через запятую.

в DM нужны субдомены
в HR субдомены НЕ нужны.

Может позже будет функция загрузки своих список с фильтром лишнего.

**Comment by antoxa5577** [comment]
Собрал маршруты для ютуба - загрузил в Keenetic, ютуб отлично работает. Но в гугле тоже впн включается, как исключить гугл?

**Reply by Ground-Zerro** [reply]
> в гугле тоже впн включается, как исключить гугл

В варианте с маршрутами довольно сложно. Даже при более точной маршрутизации по доменам к ютуб остальной гугл каким-то образом попадает в перечень узлов направляемых в туннель. youtube сам по себе неотъемлемая часть google эти два сервиса сильно интегрированы друг в друга.

Четкого рецепта не дам, только направление.
- Не агрегировать маршруты.
- Убрать из доменов `l.google.com` и `play.google.com`

Или глобально - перейти на маршрутизацию по доменам, а не маршрутам.

Больше что-то подсказать или помочь к сожалению нечем.
